src/main/java/spoon/support/sniper/internal/ElementSourceFragment.java

```diff
@@ -714,13 +715,13 @@ private void onCharSequence(CharType type, StringBuilder buff, Consumer<SourceFr
 			//detect longest match in matchers
 			StringMatcher longestMatcher = null;
 			for (StringMatcher strMatcher : matchers) {
-				if (strMatcher.isMatch(str, off)) {
+				if (strMatcher.isMatch(charArray, off)) {
 					longestMatcher = strMatcher.getLonger(longestMatcher);
 				}
 			}
 			// nothing has matched, best effort token
 			if (longestMatcher == null) {
-				consumer.accept(new TokenSourceFragment(str.toString(), TokenType.CODE_SNIPPET));
+				consumer.accept(new TokenSourceFragment(Arrays.toString(charArray), TokenType.CODE_SNIPPET));
```

Comment:
this looks like this best-effort token never worked. I guess we don't have any single test here :)?

Reply:
Looking at the coverage, there should exist a testcase for this line https://coveralls.io/builds/45751341/source?filename=src%2Fmain%2Fjava%2Fspoon%2Fsupport%2Fsniper%2Finternal%2FElementSourceFragment.java#L724. But I have not found a way in the UI to find it.

Reply:
> Looking at the coverage, there should exist a testcase for this line coveralls.io/builds/45751341/source?filename=src%2Fmain%2Fjava%2Fspoon%2Fsupport%2Fsniper%2Finternal%2FElementSourceFragment.java#L724. But I have not found a way in the UI to find it.

argh, this is https://github.com/INRIA/spoon/issues/4541

Reply:
anyway, I'd be in favor in writing an assertion on the content of this token. Otherwise, we fix broken code by something which may be buggy as well

would you be able to add such an assertion?

Reply:
After some debugging, I found the testcase https://github.com/INRIA/spoon/blob/f98e33b7c669e9c11d6a916333bcac13444b1e5d/src/test/java/spoon/test/prettyprinter/TestSniperPrinter.java#L187 using this line with a string `0,`

Reply:
great, could you write an assertion on the content of the token impacted by this PR?

Reply:
I believe the current input token is simply wrong. See for example https://www.baeldung.com/java-array-to-string.

> The default implementation of the toString() method on an array returns something like Ljava.lang.String;@74a10858 > which only informs us of the object's type and hash code.
```java
public TokenSourceFragment(String source, TokenType type) {
	this.source = source;
	this.type = type;
}
```
The source String should contain the original source token and not a string like `"[C@2d96543c"`. It's pretty hard to assert against something internal and defect.

Reply:
> I believe the current input token is simply wrong.

Agree, that's why we need a new test or a new assertion to assert the actual correct content. Would you write this?

Reply:
I can only trigger this line with joint declarations but currently the sniper is not able to work with them, see https://github.com/INRIA/spoon/issues/3386. It looks like either this token was never relevant or is needed in a future version and we can't great asserts for the value currently.

Reply:
If we are unable to specify this line, then the best option is to simply remove it. Could you give it a try?

Reply:
It looks like the token is for some strange reasons important for the sniper printing and removing it adds unwanted whitespaces. So we more or less have to keep it and fix this after fixing the linked problem.

Reply:
ack, then let's revert this last commit and proceed with the change, it's not harmful. we replace code we don't understand by code we still don't understand but less confusing.